### PR TITLE
[Ide] If MEF is queried before startup handlers are run, throw an exc…

### PR DESCRIPTION
**Pull request by Therzok**
…eption

We should prevent extensions from slowing down the IDE startup if they request the composition before startup even occurs.
Mark anything after StartupHandlers as safe and only log a warning if queried too early

Fixes VSTS #973923 - Querying MEF in startup handlers should throw

### CODE REVIEW

**Comment by Therzok** [comment]
@monojenkins backport release-8.3

**Comment by Therzok** [comment]
Okay, perfect, tests failed. Need to change it so we opt-in for exception behaviour right before addin composition and opt out when command handlers are done.

**Comment by Therzok** [comment]
Added 2 changes:
1. made the behaviour opt in and turn off after StartupHandlers are run
2. start composing mef earlier in the process initialization phase, at the same time we just start logging a warning

**Comment by Therzok** [comment]
@monojenkins backport release-8.3

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/Ide.cs [hunk 1]
```diff
@@ -314,6 +314,10 @@ static void OnInitialized ()
 			// Startup commands
 			Counters.Initialization.Trace ("Running Startup Commands");
 			AddinManager.AddExtensionNodeHandler ("/MonoDevelop/Ide/StartupHandlers", OnExtensionChanged);
+
+			// Let extensions now access CompositionManager.Instance and start asynchronously composing the catalog
+			CompositionManager.ConfigureUninitializedMefHandling (throwException: false);
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/Ide.cs]: Shouldn't this be called before the above AddinManager.AddExtensionNodeHandler call? AddExtensionNodeHandler will cause startup handlers to be executed.

Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/Ide.cs]: Yes, we don't want addins to access mef before startup handlers are run.

**Pull request merged**